components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -250,6 +254,13 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
                 associateTransientRequestData(request, responseWrapper, context);
             }
 
+            // Check if the login access to the application is enabled.
+            if (!checkIfApplicationAccessEnabled(request, responseWrapper, context)) {
```

Comment:
Lets do this as a very 1st step before we initialize anything related to the login flow.

Reply:
But in certain flows we require to have the client Id to fetch the service provider. So, We have to do it at this place since we can get the relying party from context if not from the request. WDYT?

Reply:
if you check the flow. In the beginning even for the context we set the relying party form the request. So we can do the same. It is better to do this check before anything related to login flow as if the app is disabled we don't allow anything. And this will be more optimized and we don't have unnecessary stale data

Reply:
We need to rely on the context in flows when it is a re-direction to the framework. If we only prevent at the initial request (relying on the request params) to the FW based on the disable status, if there is a case where the user has initiated the request, and the app was enabled and he/she has proceeded with next steps. In the middle if the app got disabled, still the user be able to complete the flow. IMO, this can be a fraction of those who can proceed with such a case but completely disabling is more preferable as it will ensure nobody gets the access. 
WDYT?

Reply:
It was decided after the disucssion with @darshanasbg and @janakamarasena to prevent access even at intermediate steps.